packages/rum/src/domain/record/elementsScrollPositions.ts

```diff
@@ -0,0 +1,27 @@
+export type ElementsScrollPositions = ReturnType<typeof createElementsScrollPositions>
+export type ScrollPositions = { scrollLeft: number; scrollTop: number }
+
+export function createElementsScrollPositions() {
+  const scrollPositionsByElement = new WeakMap<Element, ScrollPositions>()
+  /**
+   * The element carrying the document scroll positions is not necessarily the same on all browsers.
+   * HTML and BODY elements has been observed but other cases might exist.
+   * It should be the first serialized elements though.
+   */
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏re: “It should be the first serialized elements though.”: FMU, `set` is not necessarily called in serialization order though.

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏use [`document.scrollingElement`](https://developer.mozilla.org/en-US/docs/Web/API/document/scrollingElement) when available